### PR TITLE
updated rules with enum

### PR DESCRIPTION
**Pull request by nchiru**
added enums in below rules
lldp-session-state.rule
ethernet-oam-lfm.rule
ethernet-oam-lfm.yml
bfd-session-state.rule
evpn-instance-check.rule

### CODE REVIEW

File: juniper_official/lldp/lldp-session-state.rule [hunk 1]
```diff
@@ -67,13 +62,12 @@ healthbot {
...
                     when {
-                        matches-with "$neighbor-state" true;
-                        matches-with "$if-lldp-state" true;
+                        matches-with "$interface-lldp-state" true;
                     }
                     then {
                         status {
                             color green;
-                            message "Interface $interface-name lldp session is up";
+                            message "LLDP interface $interface-name status is Up";
```
Comment (gaddevijay) [inline juniper_official/lldp/lldp-session-state.rule]: Change this to "LLDP status on interface $interface-name is UP".
LLDP status is coming as true and we are displaying message as UP. Is this correct? Should we say that LLDP on interface is enabled or something like that?

File: juniper_official/oam/ethernet-oam-lfm.rule [hunk 2]
```diff
@@ -27,6 +27,26 @@ healthbot {
                 }
                 type string;
                 description "Discovery state for a given interface";
+                enumerate Passive Wait as 0;
+                enumerate Send Any as 1;
+                enumerate Send Local Remote as 1;
```
Comment (gaddevijay) [inline juniper_official/oam/ethernet-oam-lfm.rule]: We cannot have 2 values representing 1.

Also Send Local Remote and Send Local Remote Ok are good states or bad states? if they are bad states please keep it less than 1

File: juniper_official/oam/ethernet-oam-lfm.rule [hunk 3]
```diff
@@ -27,6 +27,26 @@ healthbot {
...
+                enumerate Send Local Remote Ok as 3;
+            }
+            field lfmd-status {
```
Comment (gaddevijay) [inline juniper_official/oam/ethernet-oam-lfm.rule]: New field not used in triggers are not stored in TSDB. Why are these being added?

File: juniper_official/routing/bfd-session-state.rule [hunk 4]
```diff
@@ -35,13 +35,19 @@ healthbot {
                 }
                 type string;
                 description "BFD session state";
+                enumerate Failing as -1;
```
Comment (gaddevijay) [inline juniper_official/routing/bfd-session-state.rule]: to states can have same value -1

File: juniper_official/vpn/evpn/evpn-instance-check.rule [hunk 5]
```diff
@@ -30,6 +30,8 @@
                 }
                 type string;
                 description "Mode of evpn Interface";
+                enumerate single-homed as 0;
```
Comment (gaddevijay) [inline juniper_official/vpn/evpn/evpn-instance-check.rule]: Is single homed a bad state?
If it is a good state it should be >=1

File: juniper_official/lldp/lldp-session-state.rule (hunk 1 above)
Comment (gaddevijay) [reply juniper_official/lldp/lldp-session-state.rule]: Also line 81 `message "Interface-$interface-name lldp session down";`
Please put a space after "Interface" and remove "-"

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/lldp/lldp-session-state.rule (hunk 1 above)
Comment (nchiru) [reply juniper_official/lldp/lldp-session-state.rule]: updated as per the comments

File: juniper_official/oam/ethernet-oam-lfm.rule (hunk 2 above)
Comment (nchiru) [reply juniper_official/oam/ethernet-oam-lfm.rule]: these are good states. updated with positive values.

File: juniper_official/routing/bfd-session-state.rule (hunk 4 above)
Comment (nchiru) [reply juniper_official/routing/bfd-session-state.rule]: updated with negative values

File: juniper_official/vpn/evpn/evpn-instance-check.rule (hunk 5 above)
Comment (nchiru) [reply juniper_official/vpn/evpn/evpn-instance-check.rule]: single homed is the default value and is good state. updated with positive value

**Review by gvarao — APPROVED** [review]

File: juniper_official/lldp/lldp-session-state.rule (hunk 1 above)
Comment (gaddevijay) [reply juniper_official/lldp/lldp-session-state.rule]: You did not answer this questions "LLDP status is coming as true and we are displaying message as UP. Is this correct? Should we say that LLDP on interface is enabled or something like that?"

Please check with Venkat and comment if this is ok or not

Comment (gaddevijay) [reply juniper_official/lldp/lldp-session-state.rule]: Line 81 message format also needs to be corrected as you have corrected message in line 70

File: juniper_official/oam/ethernet-oam-lfm.rule [hunk 6]
```diff
@@ -27,20 +27,34 @@ healthbot {
                 }
                 type string;
                 description "Discovery state for a given interface";
+                enumerate Passive Wait as 0;
+                enumerate Send Any as 1;
```
Comment (gaddevijay) [inline juniper_official/oam/ethernet-oam-lfm.rule]: Are you sure that enum names will work without double quotes?

I tried to load into terminal and getting below error.

         terminal:24:(38) syntax error: Wait
    'enumerate Passive Wait as 0;'
      syntax error

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/oam/ethernet-oam-lfm.rule [hunk 7]
```diff
@@ -27,20 +27,34 @@ healthbot {
                 }
                 type string;
                 description "Discovery state for a given interface";
+                enumerate "Passive Wait" as 0;
+                enumerate "Send Any" as 1;
```
Comment (gaddevijay) [inline juniper_official/oam/ethernet-oam-lfm.rule:31]: I just spoke to Venkat and as per him 2 and 3 are not good states. So it has to be less than 1

File: juniper_official/oam/ethernet-oam-lfm.rule [hunk 8]
```diff
@@ -27,20 +27,34 @@ healthbot {
...
+                enumerate "Send Local Remote" as 2;
+                enumerate "Send Local Remote Ok" as 3;
             }
-            field lfm-interface-name {
+            field lfmd-status {
```
Comment (gaddevijay) [inline juniper_official/oam/ethernet-oam-lfm.rule]: change `lfmd-status` to `lfm-status`

**Review by gaddevijay — CHANGES_REQUESTED** [review]

File: juniper_official/oam/ethernet-oam-lfm.rule (hunk 7 above)
Comment (gvarao) [reply juniper_official/oam/ethernet-oam-lfm.rule:31]: Sorry vijay, i misled. i just gone though document 2 and 3 are positive cases. So we can keep as is now.

**Review by gaddevijay — APPROVED** [review]

**Pull request merged**